uaclient/event_logger.py

```diff
@@ -145,23 +166,48 @@ def _generate_failed_services(self):
         }
         return list(set.union(self._failed_services, services_with_error))
 
+    def _process_events_services(self):
+        response = {
+            "_schema_version": JSON_SCHEMA_VERSION,
+            "result": "success" if not self._error_events else "failure",
+            "processed_services": sorted(self._processed_services),
+            "failed_services": sorted(self._generate_failed_services()),
+            "errors": self._error_events,
+            "warnings": self._warning_events,
+            "needs_reboot": self._needs_reboot,
+        }
+
+        print(json.dumps(response, sort_keys=True))
+
+    def _process_events_status(self):
+        output = format_machine_readable_output(self._output_content)
+        output["result"] = "success" if not self._error_events else "failure"
+        output["errors"] = self._error_events
+        output["warnings"] = self._warning_events
+
+        if self._event_logger_mode == EventLoggerMode.JSON:
+            from uaclient.util import DatetimeAwareJSONEncoder
+
+            print(
+                json.dumps(
+                    output, cls=DatetimeAwareJSONEncoder, sort_keys=True
+                )
+            )
+        elif self._event_logger_mode == EventLoggerMode.YAML:
+            import yaml
```

Comment:
Same question here

Reply:
Here the error is even weirder - but no, it does not import correctly :/

Reply:
Okay, that one is odd, what is the error that happens ?

Reply:
It fails to import yaml even before running any test. I found it weird because yaml is imported in many other places. I tried to debug this but I think I don't know too deep about the import machinery :(

I am pasting a full log here:
```
==================================================================== log start =====================================================================
Processing ./.tox/.tmp/package/2/ubuntu-advantage-tools-20.3.zip
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /home/renan/work/ubuntu-advantage-client/.tox/py3/bin/python /home/renan/work/ubuntu-advantage-client/.tox/py3/lib/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmp3jrhjff_
       cwd: /tmp/pip-req-build-zj18i51m
  Complete output (22 lines):
  Traceback (most recent call last):
    File "/home/renan/work/ubuntu-advantage-client/.tox/py3/lib/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
      main()
    File "/home/renan/work/ubuntu-advantage-client/.tox/py3/lib/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/home/renan/work/ubuntu-advantage-client/.tox/py3/lib/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 130, in get_requires_for_build_wheel
      return hook(config_settings)
    File "/tmp/pip-build-env-9hc7s_9c/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 177, in get_requires_for_build_wheel
      return self._get_build_requires(
    File "/tmp/pip-build-env-9hc7s_9c/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 159, in _get_build_requires
      self.run_setup()
    File "/tmp/pip-build-env-9hc7s_9c/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 281, in run_setup
      super(_BuildMetaLegacyBackend,
    File "/tmp/pip-build-env-9hc7s_9c/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 174, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 8, in <module>
      from uaclient import defaults, util, version
    File "/tmp/pip-req-build-zj18i51m/uaclient/util.py", line 29, in <module>
      from uaclient import event_logger, exceptions, status
    File "/tmp/pip-req-build-zj18i51m/uaclient/event_logger.py", line 12, in <module>
      import yaml
  ModuleNotFoundError: No module named 'yaml'
  ----------------------------------------

```

Reply:
Second solution seems way better - I'll check. Thanks for taking a look into this.

Reply:
So, even if we remove the `util` import from `setup.py`, it imports `version`, which in its turn imports `util` (and needs it).

I think that between defining `pyyaml` as a build-time requirement and importing it as it is, I prefer to leave it as is.
Please let me know otherwise.